SpringSecurityOauth2ProviderGrailsPlugin.groovy

```diff
@@ -470,6 +479,41 @@ class SpringSecurityOauth2ProviderGrailsPlugin {
         }
     }
 
+    private configureBasicAuthenticationFilter = { conf ->
+        // Should the basic authentication filter be registered in the filter chain
+        boolean registerBasicAuthFilter = conf.oauthProvider.registerBasicAuthenticationFilter as boolean
+
+        if(registerBasicAuthFilter) {
+            // Create a separate authentication manager with a single provider for the basic authentication filter
+            basicClientAuthenticationManager(ProviderManager) {
+                providers = [ ref('clientCredentialsAuthenticationProvider') ]
+                authenticationEventPublisher = ref('authenticationEventPublisher')
+                eraseCredentialsAfterAuthentication = conf.providerManager.eraseCredentialsAfterAuthentication // true

```

Comment:
I don't believe so. I'll change it to `eraseCredentialsAfterAuthentication = true`